### ISSUE TITLE
Implicit Many-to-Many on Delete.

### ISSUE THREAD

**Issue by Hardel-DW** [prisma/prisma#16390]
### Bug description

The bug is quite simple to understand.
To facilitate the reading of this bug report I put an example diagram.

```
model Item {
  id          Int        @id @default(autoincrement())
  categories  Category[]
  (Other field...)
}

model Category {
  id        Int       @id @default(autoincrement())
  items     Item[]
  (Other field...)
}
```

To explain how this bug appear, I will proceed step by step.
1. To start we need two tables with implicit many-to-many relationship. (Exemple: Items and Categories)
2. Now Items and Categories each have records in the database.
3. Now we link them together, with Prisma Studio for example or with PrismaClient.
4. Now in case I delete an element in either the Items or Categories table that has a relationship with the other table.
We see that in the associative table between Categories and Items that a record row is present but empty.

So basically the data is not deleted in cascade.
We end up with tons of empty lines that point to a non-existent element.

![chrome_pPFgEyf127](https://user-images.githubusercontent.com/42413417/203149853-8588376f-ca48-42d6-8552-4eb2c342cb85.png)

### How to reproduce

1. To start we need two tables with implicit many-to-many relationship. (Exemple: Items and Categories)
2. Now Items and Categories each have records in the database.
3. Now we link them together, with Prisma Studio for example or with PrismaClient.
4. Now in case I delete an element in either the Items or Categories table that has a relationship with the other table.
We see that in the associative table between Categories and Items that a record row is present but empty.

### Expected behavior

The data must be deleted if one of the two fields of the associative table is missing.

### Prisma information

```prisma
model Item {
  id          Int        @id @default(autoincrement())
  categories  Category[]
  createdAt   DateTime   @default(now())
  updatedAt   DateTime?  @updatedAt
}

model Category {
  id        Int       @id @default(autoincrement())
  items     Item[]
  createdAt DateTime  @default(now())
  updatedAt DateTime? @updatedAt
}
```

```ts
    async delete(id: number) {
        return await prisma.item.delete({
            where: {
                id
            }
        });
    }
    
    await delete(1);
```

### Environment & setup

- OS: Windows 11
- Database: PlanetScale
- Node.js version: 16.14.2
- Yarn

### Prisma Version

- Prisma and Prisma Client 4.6.1

**Comment by janpio** [comment]
I assume you are using the `referentialIntegrity` preview feature with `relationMode="prisma"`?

**Comment by Hardel-DW** [comment]
Thank you, I was not too present to answer your question.
Thank you, I will do that.

As it was indicated by PlanetScale of indicated :
"previewFeatures = ["referentialIntegrity"]" 

I added it without really understanding what it could be.

**Comment by edmbn** [comment]
I can confirm this is happening for me too. Unfortunately, I cannot get rid of   `relationMode = "prisma"` since I'm working with Planetscale.

**Comment by edmbn** [comment]
This error can be more problematic than it seems. If you retrieve from a pivot table like this:

```
select: {
  id: true,
}
```

it will return the items, although they no longer exist on the original table (not the pivot). To solve this problem, as a patch, you need to include another field that won't exist in the pivot table, for example:

```
select: {
  id: true,
  createdAt: true,
}
```

**Comment by jonasmerlin** [comment]
Yes, this is happening for me as well and causes a lot of headaches.

Thanks for the workaround @edmbn

**Comment by edmbn** [comment]
Is this issue in any roadmap version? Tests where made and there even exist a draft pull request with fixes. Looks like it was on the 4.9.0 milestone but that never happened.

**Comment by zhanghsoft** [comment]
这个问题 现在还存在

**Comment by ahkhanjani** [comment]
I just realized this is a thing the hard way. Please address this as soon as possible + provide a way to fix the existing data that have been affected by this (if even possible)

**Comment by stuartrobinson3007** [comment]
As a workaround if you're trying to delete relationships between many-to-many tables, you can do this with $executeRaw if you find the name of the generated join table (for example. by looking in Planetscale).

If you've got two tables, "item" and "category" then it'll be _ItemToCategory

The generated table has two columns, A and B. Each column is the relation id.

You can manually delete the records in the join table using $executeRaw

For example, to completely clear the join table of all records you can perform:

await prisma.$executeRaw`DELETE FROM _ItemToCategory`

Or just clear all join records for a single Category

await prisma.$executeRaw`DELETE FROM _ItemToCategory WHERE B = 'example-category-id'`

**Comment by morascode** [comment]
This is happening for my team too. It can be a source of many bugs if we are not careful with how we write our queries.

We will implement a workaround for now by disconnecting all m-n relations from a record before deleting it.

This issue has been open since over a year now, can it please be addressed soon?

**Comment by ahkhanjani** [comment]
Could a member from the Prisma team please take a look at this. This is a serious flaw.
